### PR TITLE
Sai fixes for imx8mp

### PR DESCRIPTION
**Pull request by dbaluta**
Couple of fixes to make SAI work on i.MX8MP.

### CODE REVIEW

File: src/platform/imx8m/lib/dai.c [hunk 1]
```diff
@@ -20,13 +20,13 @@ static SHARED_DATA struct dai sai[] = {
 		.base = SAI_3_BASE,
 		.fifo[SOF_IPC_STREAM_PLAYBACK] = {
 			.offset = SAI_3_BASE + REG_SAI_TDR0,
-			.depth = 16,
+			.depth = 128, /* in 4 bytes words */
```
[the remark this answers is not in the document]
Comment (dbaluta) [reply src/platform/imx8m/lib/dai.c:23]: This field is not even used.  We use SAI_FIFO_WORD_SIZE macro instead. But nevertheless it is fine to change the value here and at some point in the future we replace SAI_FIFO_WORD_SIZE macro with .depth from here.

[the remark this answers is not in the document]
Comment (paulstelian97) [reply src/platform/imx8m/lib/dai.c:23]: Uh, SDMA is reading from this field (indirectly I think). So no, this is definitely used. EDMA also reads from here on our other platforms.

Only the SAI itself may have this issue, which should be resolved soon.

[the remark this answers is not in the document]
Comment (dbaluta) [reply src/platform/imx8m/lib/dai.c:23]: Oh, i see. It is read via DAI
```
»       »       dd->config.burst_elems =
»       »       »       dd->dai->plat_data.fifo[dai->direction].depth;  

```

**Comment by dbaluta** [comment]
SOFCI TEST

**Comment by dbaluta** [comment]
@lgirdwood good to merge.

**Pull request merged**